File: data/imported/knowledge/slides/net/merged/snippets/snippet_050.cs
[Fact]
    public void TestLineDashStyle()
    {
        // Dash style persists.
        using var pres = new Presentation();
        var slide = ClearSlide(pres);
        var shape = slide.Shapes!.AddAutoShape(ShapeType.Rectangle, 50, 50, 200, 100);
        var lf = shape.LineFormat;
        lf.Width = 3;
        lf.DashStyle = LineDashStyle.Dash;
        lf.FillFormat.FillType = FillType.Solid;
        lf.FillFormat.SolidFillColor.Color = Color.Black;

        using var pres2 = TestHelpers.SaveAndReopen(pres, _tempDir);
        var lf2 = pres2.Slides[0].Shapes![0].LineFormat;
        lf2.DashStyle.Should().Be(LineDashStyle.Dash);
    }

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_050.cpp
TEST_F(ShapesIntegrationTest, MultipleShapeTypes) {
    std::vector<ShapeType> types = {ShapeType::RECTANGLE, ShapeType::ELLIPSE, ShapeType::TRIANGLE};
    Presentation pres;
    auto& slide = blank_slide(pres);
    for (auto st : types) {
        auto& s = slide.shapes().add_auto_shape(st, 10, 10, 100, 100);
        EXPECT_EQ(s.shape_type(), st);
    }
    EXPECT_EQ(slide.shapes().size(), 3);
}